### ISSUE TITLE
Fetch multiple markdown files

### ISSUE THREAD

**Issue by niklasnorin** [obedm503/bootmark#5]
Hi,

I was wondering if there's any easy way to fetch multiple .md-files into the same html-document?

What I'm basically after is a way to have multiple .md-files, say "chapter-1.md", "chapter-2.md" etc. and then have one "index.html" which pulls in the markdown from all those files and runs bootmark.

Using several bootmark tags works, but then they are completely isolated entities regarding TOC etc.

Do you have any suggestions on how to implement this, or any plans on supporting it?

Best regards,
Niklas

**Comment by obedm503** [comment]
I'm trying to figure out what the best approach to this would be...

It's either make the code that handles toc creation look thru the entire document and generate a toc from that. 
Or, allow users to pass an array of urls to the fetch property. Then all the files would be fetched, and merged to be processed as one. The toc code wouldn't change much. This might allow people to have multiple markdown files within a single element, but also allow them to have certain files be isolated from the rest, at least toc-wise.

In short,yes. I do plan to add this feature. If you have another way that might be better or easier to use, let me know.

**Comment by obedm503** [comment]
if you really want to try it out, this feature is working on the development branch. I'll keep testing it and try to solve bugs as I find them and then update the master/stable branch

**Comment by niklasnorin** [comment]
Thanks for the fast reply!

I'll take it for a test run tonight and let you know how it goes.

**Comment by niklasnorin** [comment]
@obedm503 It seems to work great, thank you!

I have it up and running here:
https://niklasnorin.github.io/thesis/

**Comment by obedm503** [comment]
great. a suggestion: I noticed that you passed `<br>` as the join string. This works fine but showdown (the markdown parser) interprets `<br>` as markdown and so it wraps it in `<p>` tags. If all you want is a line break, you can just pass nothing/empty string like so

``` html
<bootmark fetch="['file1','file2']" join="">
```

or pass in space

``` html
<bootmark fetch="['file1','file2']" join="     ">
```

bootmark automatically adds line breaks before and after the join string for safety measures.

**Comment by obedm503** [comment]
This is now on the latest stable release v[0.6.1](https://github.com/obedm503/bootmark/tree/0.6.1)
